bpf/lib/nodeport.h

```diff
@@ -1729,7 +1857,12 @@ static __always_inline int nodeport_lb4(struct __ctx_buff *ctx,
 	if (!svc || !lb4_svc_is_routable(svc)) {
 		if (svc)
 			return DROP_IS_CLUSTER_IP;
-
+#ifdef ENABLE_NAT_46X64_STATELESS
+		if (ip4->daddr != IPV4_DIRECT_ROUTING) {
+			ep_tail_call(ctx, CILIUM_CALL_IPV46_RFC8215);
+			return DROP_MISSED_TAIL_CALL;
+		}
+#endif
```

Comment:
@borkmann I didn't look too deep. But don't we want this to be in scope of the `skip_service_lookup` label below, so that ICMP traffic also gets 46-translated?

Reply:
@julianwiedmann good point, I was focused only on TCP/UDP, but our nat46x64 code also supports icmp. I'm currently working on the code anyway, so will fix it up along the way. Thanks for spotting!